util/util.go

```diff
@@ -0,0 +1,42 @@
+package util
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha512"
+	"io"
+
+	"github.com/flynn/go-tuf/data"
+)
+
+type BytesReadCloser struct {
+	*bytes.Reader
+}
+
+func (b BytesReadCloser) Close() error {
+	return nil
+}
+
+func FileMetaEqual(actual data.FileMeta, expected data.FileMeta) bool {
+	if actual.Length != expected.Length {
+		return false
+	}
+	for typ, hash := range expected.Hashes {
+		if !hmac.Equal(actual.Hashes[typ], hash) {
```

Comment:
Yeah, that sounds like a reasonable thing to do.

It seems a little heavyweight to add an error return field to indicate no intersection on available hashes, but maybe it's reasonable -- it is basically doing something like protocol compatibility probing, and that's worthy of error handling.